source/extensions/omni.isaac.lab/omni/isaac/lab/envs/utils.py

```diff
@@ -9,11 +9,92 @@
 import torch
 from typing import Any
 
-from .common import ActionType, AgentID, EnvStepReturn, ObsType, StateType, VecEnvObs, VecEnvStepReturn
+from .common import ActionType, AgentID, EnvStepReturn, ObsType, SpaceType, StateType, VecEnvObs, VecEnvStepReturn
 from .direct_marl_env import DirectMARLEnv
 from .direct_rl_env import DirectRLEnv
 
 
+def spec_to_gym_space(spec: SpaceType) -> gym.spaces.Space:
+    """Generate an appropriate Gymnasium space according to the given space specification.
+
+    Args:
+        spec: Space specification.
+
+    Returns:
+        Gymnasium space.
+
+    Raises:
+        ValueError: If the given space specification is not valid/supported.
+    """
+    if isinstance(spec, gym.spaces.Space):
+        return spec
+    # fundamental spaces
+    # Box
+    elif isinstance(spec, int):
```

Comment:
do we have any test cases that cover all of the possible types?

Reply:
Test added in https://github.com/isaac-sim/IsaacLab/pull/1117/commits/2a3dff6db4729a2ba59606991205344c5ce81ca9